### ISSUE TITLE
an issues about your ocr data iteration

### ISSUE THREAD

**Issue by zzdang** [ThomasDelteil/HandwrittenTextRecognition_MXNet#5]
Hi,your project is cool ,but your OCR_LSTM_CTC's data iteration is very slow?
Could you update it?
Thank you very much

**Comment by jonomon** [comment]
Hi,

Would it be possible for you to include some clarifications? 
Which data iteration?

**Comment by zzdang** [comment]
Your "Data Loading" module isn't iterative, you load image data and labels first time.
If the training dataset is big,the "images_data" in your "data loading" is hard to handle,the trainning will be very slow.......

**Comment by ThomasDelteil** [comment]
@zzdang, this is true, this was an acceptable trade off given the small size of the IAM dataset, and to get the ability to load pre-processed images quickly.

If your dataset is larger I would recommend using the `ImageFolderDataset` available in Gluon that would let you load each image only when necessary.

**Comment by samar-smida** [comment]
@jonomon 
@ThomasDelteil 
@ThomasDelteil
I am testing your project but I have an assertion error even though I put my email and password in the credentials.json .
In the registration form, they ask for the email and not the username.
I found this link in the project to have but it is not functional https://fki.tic.heia-fr.ch/DBs/iamDB/iLogin/index.php
please can you help me

**Comment by jonomon** [comment]
You can download the dataset https://fki.tic.heia-fr.ch/databases/iam-handwriting-database.
